src/lib/components/molecules/waffled-squares/index.jsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import { SvgSquare } from '$particles/svg-square'
+
+export const WaffledSquare = ({
+  name = 'Germany',
+  squaresTotal = 96,
+  waffleWidth = 100,
+  waffleHeight = waffleWidth,
+  rowLong = 10,
+  squareWidth = waffleWidth / rowLong,
+  labelPosX = 0,
+  labelPosY = 0,
+  groups = []
+
+}) => {
+
+  const rows = [];
+  let xPos = 0;
+  let yPos = waffleHeight;
+  let cont = 0;
+
+  const [hidden, setHidden] = useState(true);
+
+  const colours = [];
+
+  groups.map(group => {
+
+    for (let j = 0; j < group.squares; j++) {
+      colours.push(group.fill)
+    }
+
+  })
+
+  for (let i = 0; i < squaresTotal; i++) {
+
+    if (i % rowLong == 0) {
+      yPos -= squareWidth;
+      cont = 0;
+    }
+
+    xPos = cont * squareWidth;
+    cont++;
+    
+    rows.push(<SvgSquare className={'name' + i} posX={xPos} posY={yPos} width={squareWidth} fill={colours[i] ? colours[i] : "#dcdcdc"} pointerEvents={"none"} />);
+  }
+
+  let rowEnds = squaresTotal % rowLong == 0;
+
+  labelPosX = rowEnds ? 0 : xPos + squareWidth + 2;
+  labelPosY = rowEnds ? yPos - 15 : yPos;
+
+  let headerMarginBottom = squaresTotal >= (rowLong * rowLong) && waffleWidth == waffleHeight ? 15 : 4;
+
+  return (
+    <div>
+      <h2 style={{
```

Comment:
It sounds like you were already about to change this, but let's apply the styles using CSS modules.